graphql_compiler/query_pagination/pagination_planning.py

```diff
@@ -1,53 +1,175 @@
 # Copyright 2019-present Kensho Technologies, LLC.
-from collections import namedtuple
+from abc import ABC
+from dataclasses import dataclass, field
+from typing import List, Tuple
+
+from graphql import DocumentNode
 
 from ..ast_manipulation import get_only_query_definition, get_only_selection_from_ast
-from ..cost_estimation.helpers import is_int_field_type, is_uuid4_type
+from ..cost_estimation.helpers import is_uuid4_type
+from ..cost_estimation.int_value_conversion import field_supports_range_reasoning
 from ..exceptions import GraphQLError
+from ..schema.schema_info import QueryPlanningSchemaInfo
+
+
+@dataclass
+class PaginationAdvisory(ABC):
+    message: str = field(init=False)
+
+
+@dataclass
```

Comment:
nit: Any reason we don't want all of these to be frozen? Or does the `__post_init__` not play nicely with that for some reason?

Reply:
yes, post init doesn't work